src/main/scala/io/iohk/ethereum/vm/DataWord.scala

```diff
@@ -78,9 +72,14 @@ class DataWord private (private val n: BigInt) extends Ordered[DataWord] {
 
   def /(that: DataWord): DataWord = DataWord(this.n / that.n)
 
+  def **(that: DataWord): DataWord = DataWord(this.n.modPow(that.n, Modulus))
+
   def compare(that: DataWord): Int = this.n.compare(that.n)
 
-  def intValue: Int = n.intValue
+  /**
+    * @return an Int with MSB=0, thus a value in range [0, Int.MaxValue]
+    */
+  def intValue: Int = n.intValue & Int.MaxValue
```

Comment:
`Modulus > Int.MaxValue` so what do you mean by "this case is covered"?

Anyway, the `intValue` is used for accessing the byte code (usages in `Memory` are irrelevant because of gas calculation beforehand). But accessing code at undefined position should return zeroes anyway, so perhaps we shouldn't worry about this going negative, @whysoserious ?